copylot/hardware/cameras/flir/flir_camera.py

```diff
@@ -0,0 +1,524 @@
+        """
+        Return transport layer device nodemap
+        """
+        return self._nodemap_tldevice
+
+    @nodemap_tldevice.setter
+    def nodemap_tldevice(self, val):
+        """
+        Set the transport layer device
+        """
+        self._nodemap_tldevice = val
+
+    @property
+    def device_id(self):
+        """
+        Return the serial number of the current camera (type: String)
+        """
+        return self._device_id
+
+    @device_id.setter
+    def device_id(self, val):
+        """
+        Set the serial number of the current camera
+        """
+        self._device_id = val
+
+    def open(self, index=0):
+        """
+        Open the system before any method can be used. Set a pointer CameraPtr to one camera.
+
+        Parameters
+        ----------
+        index: index of camera assigned to CameraPtr. Type: int
+        """
+        if self.system is None:
+            self.system = PySpin.System.GetInstance()
+            self.cam_list = self.system.GetCameras()
+        self.cam = self.cam_list[index]
+        self.nodemap_tldevice = self.cam.GetTLDeviceNodeMap()
+
+        # assign serial number
+        serial_no = ''
+        node_serial_no = PySpin.CStringPtr(
+            self.nodemap_tldevice.GetNode('DeviceSerialNumber')
+        )
+        if PySpin.IsReadable(node_serial_no):
+            serial_no = node_serial_no.GetValue()
+        else:
+            logger.error('Node serial number is not readable')
+        self._device_id = serial_no
+
+        # initialize camera
+        self.initialize()
+
+    def initialize(self):
+        """
+        Initialize the current camera. Used for settings or to start imaging.
+        """
+        try:
+            if not self.cam.IsInitialized():
+                self.cam.Init()
+        except PySpin.SpinnakerException as ex:
+            logger.error('Error running camera initialization: %s' % ex)
+
+    def close(self):
+        """
+        Close the system and delete pointer to current camera after imaging to avoid Spinnaker::Exception [-1004]
+        Call open() to image again with this instance
+        """
+        # Deinitialize camera
+        self.cam.DeInit()
+
+        # Clean CameraPtr
+        self.cam = None
+        # Clear camera list
+        self.cam_list.Clear()
+        # Release system
+        self.system.ReleaseInstance()
+
+        # set back to default
+        self.system = None
+        self.cam_list = None
+        self.device_id = None
+        self.nodemap_tldevice = None
+
+    def list_available_cameras(self):
+        """
+        Return the list (type CameraList) of all the cameras in the system
+        """
+        return self.cam_list
+
+    def save_image(self, array, n, serial_no):
+        """
+        Return the nth image data array from an initialized camera.
+
+        Parameters
+        ----------
+        n: number of images to take in that period of camera initialization. Type: int
```

Comment:
these parameter docstrings are not numpy style. It should the following convention:
```
parameter_name : parameter_type
    Detailed description about the parameter
```